testlapack/matgen.go

```diff
@@ -308,6 +317,323 @@ func dlarnv(dst []float64, dist int, rnd *rand.Rand) {
 	}
 }
 
+// dlattr generates an n×n triangular test matrix A with its properties uniquely
+// determined by imat and uplo, and returns whether A has unit diagonal. If diag
+// is blas.Unit, the diagonal elements are set so that A[k,k]=k.
+//
+// trans specifies whether the matrix A or its transpose will be used.
```

Comment:
Note that trans is not currently used or add the final transpose when asked for.